pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -94,3 +96,39 @@ func (p *v1PodResourcesServer) GetAllocatableResources(ctx context.Context, req
 		Memory:  p.memoryProvider.GetAllocatableMemory(),
 	}, nil
 }
+
+// Get returns information about the resources assigned to a specific pod
+func (p *v1PodResourcesServer) Get(ctx context.Context, req *v1.GetPodResourcesRequest) (*v1.GetPodResourcesResponse, error) {
+	metrics.PodResourcesEndpointRequestsTotalCount.WithLabelValues("v1").Inc()
+	metrics.PodResourcesEndpointRequestsGetCount.WithLabelValues("v1").Inc()
+
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.KubeletPodResourcesGet) {
+		metrics.PodResourcesEndpointErrorsGetCount.WithLabelValues("v1").Inc()
+		return nil, fmt.Errorf("Pod Resources API Get method disabled")
+	}
+
+	pod, exist := p.podsProvider.GetPodByName(req.PodNamespace, req.PodName)
+	if !exist {
+		metrics.PodResourcesEndpointErrorsGetCount.WithLabelValues("v1").Inc()
+		return nil, fmt.Errorf("Pod %s in namespace %s not found", req.PodName, req.PodNamespace)
+	}
+	podResources := &v1.PodResources{
+		Name:       pod.Name,
+		Namespace:  pod.Namespace,
+		Containers: make([]*v1.ContainerResources, len(pod.Spec.Containers)),
+	}
+
+	for i, container := range pod.Spec.Containers {
+		podResources.Containers[i] = &v1.ContainerResources{
+			Name:             container.Name,
+			Devices:          p.devicesProvider.GetDevices(string(pod.UID), container.Name),
+			CpuIds:           p.cpusProvider.GetCPUs(string(pod.UID), container.Name),
+			Memory:           p.memoryProvider.GetMemory(string(pod.UID), container.Name),
+			DynamicResources: p.dynamicResourcesProvider.GetDynamicResources(pod, &container),
+		}
+	}
+
+	return &v1.GetPodResourcesResponse{
+		PodResources: podResources,
+	}, nil
```

Comment:
please split into:
```
response := &v1.GetPodResourcesResponse{
	PodResources: podResources,
}

return response, nil
```

Reply:
it will be not consistent with other return calls see [1]. Do you want me to change them as well?


[1]- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/podresources/server_v1.go#L91-L94

Reply:
Ugh. OK, better to stay consistent. We can change all of them as a follow-up.